### PR TITLE
Mail configuration provider

### PR DESCRIPTION
**Pull request by BorisNen**
This PR is related to https://github.com/SAP/xsk/issues/899

Changelog:
1. MailFacade's default instance of the MailClient class now fetches all properties through the mail provider service
2. The mail provider service can be selected through an env variable - DIRIGIBLE_MAIL_CONFIG_PROVIDER
3. Added missing mail env variables to Configuration

### CODE REVIEW

File: modules/mail/mail-env-config/src/main/java/org/eclipse/dirigible/mail/env/EnvMailConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        addValue(properties, key, envKey, null);
+    }
+
+    private static void addValue(Properties properties, String key, String envKey, String defaultValue) {
+        String value = Configuration.get(envKey);
+        if (value != null) {
+            properties.put(key, value);
+        } else if (defaultValue != null) {
+            properties.put(key, defaultValue);
+        }
```
Comment (ThuF) [inline modules/mail/mail-env-config/src/main/java/org/eclipse/dirigible/mail/env/EnvMailConfigProvider.java:91]: Could we use here `Configuration.setIfNull()`?

**Review by ThuF — COMMENTED** [review]
@BorisNen +1, Looks good to me, just one small comment, before merging the PR.

File: modules/mail/mail-env-config/src/main/java/org/eclipse/dirigible/mail/env/EnvMailConfigProvider.java (hunk 1 above)
Comment (BorisNen) [reply modules/mail/mail-env-config/src/main/java/org/eclipse/dirigible/mail/env/EnvMailConfigProvider.java:91]: Configuration.setIfNull() is a void method that sets key-value pairs inside Configuration itself.
Here, we want to check if we should put the returned or default value inside the Properties instance.

Comment (ThuF) [reply modules/mail/mail-env-config/src/main/java/org/eclipse/dirigible/mail/env/EnvMailConfigProvider.java:91]: Agreed, thanks for the update! 😄

**Pull request merged**